ERCS/erc-7844.md

```diff
@@ -0,0 +1,529 @@
+
+
+### Interfaces
+```solidity
+    function init_create(
+    uint256[] memory types, 
+    uint256[] memory sizes
+) external;
+
+    function insert_new_member(
+    uint256 valType, 
+    uint256 size, 
+    uint256 storageSpace
+) external;
+
+function push(
+	uint256 storageSpace
+) external;
+
+function pushMany(
+    uint256 amount, 
+    uint256 storageSpace
+) external;
+
+function put(
+    uint256 data, 
+    uint256 memberIndex, 
+    uint256 entryIndex, 
+    uint256 storageSpace
+) external;
+
+function put_string(
+	string memory data, 
+	uint256 memberIndex, 
+	uint256 entryIndex, 
+	uint256 storageSpace
+) external;
+
+function get(
+    uint256 memberIndex, 
+    uint256 entryIndex, 
+    uint256 storageSpace
+) external view returns(uint256);
+
+function get_string(
+    uint256 memberIndex, 
+    uint256 entryIndex, 
+    uint256 storageSpace
+) external view returns(string memory returnValue);
+
+function set_permissions(
+	uint256 level,
+	address recipient
+) external;
+
+function get_storage_space_state_data(
+    uint256 storageSpace
+) external view returns(
+	uint256 members, 
+	uint256 entries,
+	uint256 stringIndex,
+	uint256 safeIndex
+);
+
+function total_members(
+    uint256 storageSpace
+) external view returns(uint256);
+
+/**
+* For string, bitCount->stringIndex, size param is extraneous.
+*/
+function get_member_data(
+	uint256 memberIndex,
+	uint256 storageSpace
+) external view returns(
+	uint256 bitCount, 
+	uint256 valSize,
+	uint256 valType, 
+);
+
+function get_permission_level(
+	address target
+) external view returns(uint256);
+
+```
+
+### Optional Interface Members
+```solidity 
+function put_batch(
+   uint256[] memory values, 
+   uint256[] memory members, 
+   uint256 entryIndex, 
+   uint256 storageSpace
+) external;
+
+function get_batch(
+    uint256[] memory members, 
+    uint256 entryIndex, 
+    uint256 storageSpace
+) external view returns(uint256[] memory result);
+
+function get_index_from_address(
+	address indexAddress,
+	uint256 memberIndex,
+	uint256 storageSpace
+) external view returns(uint256 entryIndex);
+
+function map_address_to_index(
+	address indexAddress,
+	uint256 storageSpace,
+	uint256 targetIndex
+) internal;
+
+function put_with_address(
+	address indexAddress,
+	uint256 storageSpace,
+) external;
+
+function _get_root_slot(
+	uint256 storageSpace
+) internal view returns(uint256);
+
+```
+### Initialization
+
+![Basic Interaction Flow](../assets/erc-7844/BasicInteractionFlow.svg)
+![CDS Init, Extension](../assets/erc-7844/CDSInitExpansion.svg)  
+The `init_create` function **MUST** handle storage space creation in both an initialization and live extension setting.  The function **MUST** take an array of types and sizes as input, which **MUST** conform to the above specifications, and be equivalent in length when input.  Developers **MUST** include length equivalency validation for `types` and `sizes`.
+
+```solidity
+function init_create(
```

Comment:
You should move implementations into the reference implementation section (unless there is _exactly_ one possible implementation). The specification section should define the observable behavior of an implementation, not the internals of how it works.